pkg/pulumiyaml/ast/expr.go

```diff
@@ -741,7 +744,32 @@ func tryParseFunction(node *syntax.ObjectNode) (Expr, syntax.Diagnostics, bool)
 	default:
 		var diags syntax.Diagnostics
 		k := kvp.Key.Value()
-		if strings.HasPrefix(k, "Fn::") {
+		// Fn::Invoke can be called as Fn::{pkg}:{Name}
+		if match, _ := regexp.MatchString(fnInvokeRegexPattern, k); match {
```

Comment:
Can we address the returned error here? We should at least leave a comment saying what the error will be.

Reply:
I believe the error will be if our regex pattern can't be parsed, which should be handled by `regex.MustCompile(fnInvokeRegexPattern)` – I'll add a comment